### PR TITLE
feat(cloudprofiler): generate library

### LINKED ISSUE #8179: Stackdriver Profiler API: `cloudprofiler.googleapis.com`

### PR DESCRIPTION
**Pull request by dbolduc**
Fixes #8179 

The quickstart runs, but it doesn't actually do anything.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8425)

### CODE REVIEW

**Review by coryan — CHANGES_REQUESTED** [review]
For other services with "cloud" in the name we decided that `google/cloud/profiler` was cloudy enough.  Please rename the library to avoid excessive cloudiness.

File: google/cloud/cloudprofiler/README.md [hunk 1]
```diff
@@ -3,7 +3,15 @@
 :construction:
 
 This directory contains an idiomatic C++ client library for the
-[Cloud Profiler API][cloud-service-docs], a service to Manages continuous profiling information.
+[Cloud Profiler API][cloud-service], a service that manages continuous CPU
+and heap profiling to improve performance and reduce costs.
+
+Note that this library allows you to interact with Cloud Profiler, but Cloud
+Profiler does not yet offer profiling of C++ libraries. The
```
[the remark this answers is not in the document]
Comment (dbolduc) [reply google/cloud/cloudprofiler/README.md]: Done.

File: google/cloud/cloudprofiler/doc/main.dox [hunk 2]
```diff
@@ -2,8 +2,16 @@
...
 @mainpage Cloud Profiler API C++ Client Library
 
-An idiomatic C++ client library for the [Cloud Profiler API][cloud-service-docs], a service
-to Manages continuous profiling information.
+An idiomatic C++ client library for the [Cloud Profiler API][cloud-service-docs],
+a service that manages continuous CPU and heap profiling to improve performance
+and reduce costs.
+
+Note that this library allows you to interact with Cloud Profiler, but Cloud
+Profiler does not yet offer profiling of C++ libraries. The
```
[the remark this answers is not in the document]
Comment (dbolduc) [reply google/cloud/cloudprofiler/doc/main.dox]: Done.

**Comment by dbolduc** [comment]
> For other services with "cloud" in the name we decided that `google/cloud/profiler` was cloudy enough. Please rename the library to avoid excessive cloudiness.

Done.

**Review by coryan — APPROVED** [review]
:lgtm:

> Reviewed 43 of 43 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8425#-:-MwT8UKaABOqSbsgduH3:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @dbolduc)

**Pull request merged**